src/formatDate.js

```diff
@@ -8,7 +8,61 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splitElement = fromFormat[3];
+  const joinElement = toFormat[3];
+  const newDate = date.split(splitElement);
+
+  let formatAgan = [];
+
+  let year = formatAgan[0];
+  let month = formatAgan[1];
+  let day = formatAgan[2];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === 'YY' || fromFormat[i] === 'YYYY') {
+      year = newDate[i];
+    }
+
+    if (fromFormat[i] === 'YYYY') {
+      year = newDate[i];
+    }
+
+    if (fromFormat[i] === 'MM') {
+      month = newDate[i];
+    }
+
+    if (fromFormat[i] === 'DD') {
+      day = newDate[i];
+    }
+  }
+
+  if (toFormat.includes('YYYY') && year.length === 2 && year < 30) {
```

Comment:
The condition `year.length === 2` is always true due to the previous code structure. The `year` variable is assigned from `newDate[i]` where `i` is the index of 'YY' or 'YYYY' in `fromFormat`, and thus will always have length 2 or 4. You should explicitly check if the original format was 'YY' before assuming the length is 2.